core/services/pipeline/task.hexencode_test.go

```diff
@@ -0,0 +1,99 @@
+package pipeline_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+func TestHexEncodeTask(t *testing.T) {
+	t.Parallel()
+	bigTwo, bigThree := big.NewInt(2), big.NewInt(3)
+
+	tests := []struct {
+		name   string
+		input  interface{}
+		result string
+		error  string
+	}{
+
+		// success integers
+		{"zero", 0, "0x0", ""},
+		{"small int", 1, "0x1", ""},
+		{"two-byte integer", 256, "0x100", ""},
+		{"uint8", uint8(10), "0xa", ""},
+		{"small int64", int64(456), "0x1c8", ""},
+		{"large int64", int64(999000000000), "0xe8990a4600", ""},
+		{"bigint 1", bigTwo.Exp(bigTwo, big.NewInt(100), nil), "0x10000000000000000000000000", ""},
+		{"bigint 2", bigThree.Exp(bigThree, big.NewInt(100), nil), "0x5a4653ca673768565b41f775d6947d55cf3813d1", ""},
+		{"decimal type but integer value", 1.0, "0x1", ""},
+		{"decimal type but integer value zero", 0.0, "0x0", ""},
+
+		// success strings/bytes
+		{"string ascii bytes", "xyz", "0x78797a", ""},
+		{"string with whitespace", "1 x *", "0x312078202a", ""},
+		{"string shouldn't convert to int", "456", "0x343536", ""},
+		{"don't detect hex in string", "0xff", "0x30786666", ""},
+		// *** TODO: are we OK with leading zeros? this one is inconsistent with integer cases above
```

Comment:
@pinebit what do you think about this?

Reply:
If the leading zeroes are in integers, then we ignore them. That's what most users would expect.
If the leading zeroes prepend a string/bytes, then we don't care / don't strip them, maybe those are intentional and that's what most users would expect. IMO.

Reply:
This is about leading zeros in output, not input.

Reply:
I'll leave it as-is and change this TODO comment into a "NOTE"

Reply:
Got it, I believe the current behavior is correct. Leaving this note is good.